docs/proposals/update_replace_functions.md

```diff
@@ -0,0 +1,76 @@
+# Proposal: Improve the replacement functions
+
+_Owners_: @daneshk @kalaiyarasiganeshalingam  
+_Reviewers_: @daneshk  
+_Created_: 2022/03/11   
+_Updated_: 2022/03/11  
+_Issues_: [#2772](https://github.com/ballerina-platform/ballerina-standard-library/issues/2772) 
+
+## Summary
+The existing replacement functions don't support the following functionalities:
+- There is no way to replace the matches from the particular string index of the given string.
+- There isn't any mechanism to replace the matches with a dynamic value.
+
+So, this proposal improves the replacement functions to support the above functionalities.
+
+## Goals
+
+- Provide a way to support the above functionalities.
```

Comment:
@kalaiyarasiganeshalingam shall we make new lines and space consistent across the proposal.